### PR TITLE
Refactored ControllerShould + new test with 2 sets of commands

### PR DESCRIPTION
**Pull request by lindajohnstone**
All tests passing

### CODE REVIEW

File: MarsRover.Tests/ControllerShould.cs [hunk 1]
```diff
@@ -6,67 +6,84 @@ namespace MarsRover.Tests
...
             var expectedString = "Rover can't move. Obstacle at 0,0.";
-            var controller = new Controller(mockInput.Object, output, fileMapInput);
+            
+            _controller.Run();
+
+            _output.OutputList.Should().Contain(expectedString);
+        }
+
+        [Fact]
+        public void Run_ReturnsEndMessage_Given2SetsOfRoverCommands()
```
Comment (kedoodle) [inline MarsRover.Tests/ControllerShould.cs:73]: The test should probably care about the final state of the map/rover i.e. given this map + set of commands, where do we expect the rover to be at the end?

File: MarsRover.Tests/ControllerShould.cs [hunk 2]
```diff
@@ -6,67 +6,84 @@ namespace MarsRover.Tests
 {
     public class ControllerShould
     {
+        Controller _controller;
```
Comment (indianbread) [inline MarsRover.Tests/ControllerShould.cs:9]: are you missing access modifiers here? e.g. `private readonly Controller _controller;`

File: MarsRover.Tests/ControllerShould.cs [hunk 3]
```diff
@@ -6,67 +6,84 @@ namespace MarsRover.Tests
...
+        StubOutput _output;
+        FileMapInput _fileMapInput;
+
+        public ControllerShould()
+        {
+            _mockInput = new Mock<IInput>();
+            _output = new StubOutput();
+            _fileMapInput = new FileMapInput();
+            _controller = new Controller(_mockInput.Object, _output, _fileMapInput);
+        }
```
Comment (kedoodle) [inline MarsRover.Tests/ControllerShould.cs:20]: nice

File: MarsRover.Tests/ControllerShould.cs [hunk 4]
```diff
@@ -6,67 +6,84 @@ namespace MarsRover.Tests
...
         [Fact]
         public void Run_ReturnsMapOutput_GivenRoverCommandsThatResultInNoObstacle()
         {
-            var mockInput = new Mock<IInput>();
-            mockInput.SetupSequence(_ => _.ReadLine())
+            _mockInput.SetupSequence(_ => _.ReadLine())
```
Comment (kedoodle) [inline MarsRover.Tests/ControllerShould.cs:24]: probably worth using something like `i` for input or an arbitrary character other than a discard (`_`) which implies that it won't be used

**Review by kedoodle — APPROVED** [review]

**Review by indianbread — APPROVED** [review]

File: MarsRover.Tests/ControllerShould.cs (hunk 1 above)
Comment (lindajohnstone) [reply MarsRover.Tests/ControllerShould.cs:73]: Is it the name you're referring to? i.e Run_ReturnsEndMessage_GiveMapAnd@SetsOfRoverCommands
I didn't think I should check the Rover's final location - testing implementation not behaviour.

File: MarsRover.Tests/ControllerShould.cs (hunk 2 above)
Comment (lindajohnstone) [reply MarsRover.Tests/ControllerShould.cs:9]: I thought that if there is no modifier, plus (being a field) the name began with an underscore, it was private? To the best of my knowledge, I have followed this convention throughout this kata (& probably the last few).

File: MarsRover.Tests/ControllerShould.cs (hunk 1 above)
Comment (indianbread) [reply MarsRover.Tests/ControllerShould.cs:73]: @lindajohnstone testing the rover's final location here is not testing implementation details. Think of the commands you give as the input, and the rover's final location as the output. Given a set of commands, if your application is working correctly, the rover is expected to end up at an expected location of the given map. The end message is not a good indicator of your application working correctly - how do you know your run loop has executed correctly without checking that the rover is where it's supposed to be?

**Pull request merged**